node/router/src/lib.rs

```diff
@@ -474,6 +474,8 @@ impl<N: Network> Router<N> {
                 self.connected_peers.write().await.remove(&peer_ip);
                 // Add an entry for this `Peer` in the candidate peers.
                 self.candidate_peers.write().await.insert(peer_ip);
+                // Send a disconnect message to the peer.
+                self.handle_send(peer_ip, Message::Disconnect(DisconnectReason::NoReasonGiven.into())).await;
```

Comment:
Why are we attempting to send a disconnect right after we deque'd them from the connected peer list? This message will never be sent.

Reply:
I've removed this logic.